API/Async.pm

```diff
@@ -367,51 +395,25 @@ sub getLatestCollectionTimestamp {
 
 	my $userId = $self->userId || return $cb->();
 
-	$self->_get("/users/$userId/favorites/$type", sub {
+	$self->_get("/users/$userId/favorites", sub {
 		my $result = shift;
-
-		my $latestUpdate;
-		eval {
-			if ($type eq 'playlists') {
-				my $items = $result->{items};
-
-				my $timestamp = reduce {
-					my $ta = ref $a ? maxstr($a->{created}, $a->{item}->{lastUpdated}) : $a;
-					my $tb = ref $b ? maxstr($b->{created}, $b->{item}->{lastUpdated}) : $b;
-					maxstr($ta, $tb);
-				} @$items;
-
-				$latestUpdate = str2time($timestamp) || 0;
-			}
-			else {
-				$latestUpdate = str2time($result->{items}->[0]->{created}) || 0;
-			}
-		};
-
-		($@ || !$latestUpdate) && $log->error("Failed to get '$type' metadata: $@");
-
-		$cb->($latestUpdate);
-	},{
-		order => 'DATE',
-		orderDirection => 'DESC',
-		limit => 4,
-		_nocache => 1,
-	});
+		my $key = 'updatedFavorite' . ucfirst($type);
+		# we always return as well the timestamp of home-made playlists
+		$cb->( str2time($result->{$key}) || 0, str2time($result->{updatedPlaylists}) || 0 );
+	}, { _nocache => 1 });
```

Comment:
Would we get all timestamps for all kinds of favorites in one request? This could greatly simplify `Importer->needsUpdate()`: that currently is calling `getLatestCollectionTimestamp()` individually for each type. We could do a single call instead and check all of them?

Reply:
Yes we get TS for everything, including one for the our favorites playlists, did the list change (not the content of the playlists) but a different TS for our own playlist which indicates (AFAIU) that the *content* of our playlists has changed.